File: Leetcode/Questions/String/Sort_Characters_By_Frequency.cpp
#include <bits/stdc++.h>
using namespace std;
string frequencySort(string s)
{
    string ans = "";
    map<char, int> mp;
    int n = s.length();
    for (int i = 0; i < n; i++)
    {
        mp[s[i]]++;
    }
    vector<pair<char, int>> vec(mp.begin(), mp.end());

    sort(vec.begin(), vec.end(), [](const auto &a, const auto &b)
         { return a.second > b.second; });

    for (const auto &pair : vec)
    {
        n = pair.second;
        int j = 0;
        while (j < n)
        {
            ans += pair.first;
            j++;
        }
    }
    return ans;
}
int main()
{
    string s = "aaabbBBccc";
    cout << frequencySort(s);

    return 0;
}

File: Leetcode/Questions/String/Sort Characters By Frequency.js
/**
 * @param {string} s
 * @return {string}
 */
var frequencySort = function (s) {
  let ans = "";
  let n = s.length;
  let mp = new Map();

  for (let i = 0; i < n; i++) {
    if (mp.has(s[i])) {
      mp.set(s[i], mp.get(s[i]) + 1);
    } else {
      mp.set(s[i], 1);
    }
  }

  let arr =  Array.from(mp.entries());

  arr.sort((a, b) => b[1] - a[1]);

  for (const [char, count] of arr) {
    ans = ans + char.repeat(count);
  }

  return ans;
};
